File: runtime/src/de/fatox/meta/injection/Singleton.java
package de.fatox.meta.injection;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Identifies a type that the injector only instantiates once. Not inherited.
 */
@Scope
@Documented
@Retention(RUNTIME)
public @interface Singleton {
}

File: runtime/src/main/kotlin/de/fatox/meta/injection/Singleton.kt
package de.fatox.meta.injection

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Identifies a type that the injector only instantiates once. Not inherited.
 *
 * @see javax.inject.Scope @Scope
 */
@Scope
@Documented
@Retention(RetentionPolicy.RUNTIME)
annotation class Singleton 